Lib/test/test_decimal.py

```diff
@@ -1185,6 +1185,16 @@ def test_wide_char_separator_decimal_point(self):
         self.assertEqual(format(Decimal('100000000.123'), 'n'),
                          '100\u066c000\u066c000\u066b123')
 
+    def test_decimal_from_float_repr(self):
```

Comment:
This is just nitpicking, but I think we're testing the wrong thing here: the change is in the type of the object that's passed to the `Decimal` subclass; the `repr` of the value was just a convenient way of demonstrating the difference in behaviours between the Python and C decimal implementations. How about recording and checking the type instead of the repr?